xdsl/irdl.py

```diff
@@ -1051,11 +1065,21 @@ def new_verifier(verifier, op):
                 new_attrs["verify_"])
 
     def builder(cls,
-                operands=[],
-                result_types=[],
-                attributes=dict(),
-                successors=[],
-                regions=[]):
+                operands=None,
```

Comment:
(This is an understanding question) Are you making these changes because default arguments in python are dangerous? If so, could you provide a test case, where this actually fails/leads to non-intuitive behavior?

IMO, this is feels like a change in functionality that can be justified, but probably not as part of a documentation and type hints PR.

Reply:
If I recall, these default arguments weren't mutated, so they were perfectly valid.
However, Python linters will often complain about using these, since this is indeed quite dangerous.
So I'm fine changing these arguments, to avoid having weird problems in the future.